pyvcg/smt.py

```diff
@@ -667,6 +702,13 @@ def walk(self, visitor):
         assert isinstance(visitor, Visitor)
         return visitor.visit_record_declaration(self)
 
+    def is_recursive(self):
```

Comment:
I do not think sequences should be allowed to recurse on types used (a sequence X with element type X makes no sense).

If you re-base on my no-recursion patch, then for sequences (and parametric types in general) you should not need to do anything.